qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,437 @@
+# This Python file uses the following encoding: utf-8
+# Loick Le Guevel, 2019
+# Etienne Dumur <etienne.dumur@gmail.com>, 2021
+
+from typing import Union, Tuple
+from functools import partial
+from math import ceil
+
+from qcodes.instrument.base import Instrument
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+from qcodes.instrument.channel import MultiChannelInstrumentParameter
+from qcodes.instrument.visa import VisaInstrument
+
+
+class iTestChannel(InstrumentChannel):
+    """
+    A single channel of iTest.
+    """
+    
+    def __init__(self, parent: Instrument,
+                       name: str,
+                       chan_num: int) -> None:
+        """
+        Args:
+            parent: The instrument to which the channel is attached.
+            name: The name of the channel.
+            chan_num: The number of the channel in question
+        """
+        super().__init__(parent, name)
+        
+        self.chan_num = chan_num
+        # Get channel id
+        i, c = round(chan_num/4.)+1, chan_num%4
+        self.chan_id = 'i{};c{};'.format(i,c)
+
+        self.add_parameter('v',
+                           label='Channel {} voltage'.format(chan_num),
+                           unit='V',
+                           docstring='Voltage of the channel in volt.',
+                           get_cmd=partial(self._parent._get_voltage, chan_num),
+                           set_cmd=partial(self._parent._set_voltage, chan_num),
+                           get_parser=float
+                           )
+        
+        self.add_parameter('v_and_trig',
```

Comment:
a curiosity/design question - why is it such that the `v` parameter that has the more user-friendly name (similary to other qcodes drivers with `volt` and `voltage` parameters) does NOT actually set the voltage to that value? i'd expect that the v parameter DOES set the voltage and there's some target_v parameter that allows to tell the instrument about one's intent to set the voltage to some value.

another remark - once a voltage is being set via v_and_trig parameter, shouldn't parameter v be updated to have an invalid cache (see `.cache.invalidate` for more info), or even set to some neautral value?

Reply:
Hi,

Thanks for the remarks.
The trigger story is relative to how the ITest is built.
1. you set a ramp mode.
2. you set a ramp speed.
3. you set a target voltage.
4. you trigger the ramping.

However I discussed with a colleague and he had input that I am going to integrate and come back with a new version.
I will also change the naming to target_voltage which is maybe more meaningful.